solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -245,6 +245,15 @@ pub mod restaking {
         Ok(())
     }
 
+    pub fn update_admin(ctx: Context<UpdateStakingParams>, new_admin: Pubkey) -> Result<()> {
```

Comment:
is this properly guarded so that can only be called by the existing adming?

Reply:
I am using the existing `UpdateStakingParams` struct which was does checks if it is called by the current admin. Same method which existed when the contract was audited.
https://github.com/ComposableFi/emulated-light-client/blob/de7441e36c58f072724939fce772fe97c632a9f0/solana/restaking/programs/restaking/src/lib.rs#L563